functions.cookies.php

```diff
@@ -0,0 +1,208 @@
+<?php
+/**
+ * This file is meant to be the home for any function handling cookies that can
+ * be accessed anywhere within Jetpack.
+ *
+ * This file is loaded whether or not Jetpack is connected to WP.com.
+ *
+ * @package Jetpack
+ */
+
+/**
+ * Only proceed if Jetpack is connected to WordPress.com,
+ * and there is no active shortcircuit filter. Note that because this file
+ * executes on plugin load (to be able to define `wp_set_auth_cookie` before
+ * `pluggable.php`), a third party can only use the short-circuit filter in plugins
+ * loaded before Jetpack, or in an mu-plugin.
+ */
+if ( ! Jetpack::is_active() || apply_filters( 'jetpack_disable_auth_cookie_plugable', false ) ) {
```

Comment:
We'll need to add a docblock for that new filter, so our parser can add it to the Codex.